### PR TITLE
FEAT: Add Home page

### PR DESCRIPTION
**Pull request by VishalPatil18**
### 📝 Description

The home page is the landing page. It is the first page that the user sees when the site loads.

### 🚀 New behavior

The home page has various sections like;
- Side drawer for menu and links
- FilterTags Section
- Category Carousel
- Playlist Carousel
- Trending Videos Section
- Watch later Videos Section
- Featured Videos Section

### 🖼 Preview Link

[https://vistream-react-version-fvzxbwkhv-vishalpatil18.vercel.app/](https://vistream-react-version-fvzxbwkhv-vishalpatil18.vercel.app/)

### 💣 Is this a breaking change?
- [ ] Yes
- [x] No

### 🔬 Is the code self-reviewed?
- [x] Yes
- [ ] No

### 📋 Is the code well-formatted?
- [x] Yes
- [ ] No

### 🔗 Related to Issue #1 

> NOTE: The site is not responsive yet. I would work on responsiveness at the end.

### CODE REVIEW

**Review by starver20 — APPROVED** [review]
Nice UI, very detailed info on video cards. Code looks good to merge.

File: src/components/FilterTags/FilterTags.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const filters = [
+  "All",
+  "New",
+  "Under 4 minutes",
+  "4-20 minutes",
+  "Over 20 minutes",
+  "Most Viewed First",
+  "Oldest First",
+];
+import styles from "./FilterTags.module.css";
```
Comment (kart-c) [inline src/components/FilterTags/FilterTags.jsx:10]: keep imports at the top

**Review by kart-c — APPROVED** [review]
Clean code @VishalPatil18 
I like that you have made an icons.js so that you do need to repeat long name classes.
Also the UI looks great.
I can see that you are going to implement a carousel for your playlists and categories section, you can provide some buttons (even if you implement the functionality later it's okay). This will tell the reviewer there is going to be a carousel.

File: src/components/FilterTags/FilterTags.jsx (hunk 1 above)
Comment (VishalPatil18) [reply src/components/FilterTags/FilterTags.jsx:10]: Surely, I'll fix it in the next PR

**Pull request merged**